packages/core/src/cli/template.test.ts

```diff
@@ -0,0 +1,66 @@
+import { describe, it, expect } from 'vitest'
+import { temporary } from '@gestaltjs/testing'
+import { ScaffoldOptions, scaffold } from './template'
+import { join as joinPath } from './path'
+import { writeFile, exists, readFile, mkDir } from './fs'
+
+describe('scaffold basic file', () => {
+  it('scaffold template', async () => {
+    await temporary.directory((tmpDir) => {
+      // Given
+      const sourceDirectory = joinPath(tmpDir, 'source')
+      const targetDirectory = joinPath(tmpDir, 'target')
+      const handlebarData = {
+        name: 'my-cool-project',
+      }
+      const fileName = 'hello-world.txt'
+      const sourceFile = joinPath(sourceDirectory, fileName)
+      mkDir(sourceDirectory)
+      writeFile(sourceFile, '')
+      const scaffoldOptions: ScaffoldOptions = {
+        sourceDirectory: sourceDirectory,
+        targetDirectory: targetDirectory,
+        data: handlebarData,
+      }
+      // When
+      scaffold(scaffoldOptions)
+      // Then
+      const targetFile = joinPath(targetDirectory, fileName)
+      expect(exists(targetFile))
```

Comment:
`exists` is also [async](https://github.com/gestaltjs/gestalt/blob/main/packages/core/src/cli/fs.ts#L11) and therefore we need an await here. We can also write:

```js
await expect(exists(targetFile)).resolves.toEqual(true)
```